uibutton/Tiltfile

```diff
@@ -0,0 +1,50 @@
+def cmd_button(name, resource, argv, text=''):
+  if config.tilt_subcommand == 'down':
+    return
+
+  if not text:
+    text = name
+
+  button = {
+    "apiVersion": "tilt.dev/v1alpha1",
+    "kind": "UIButton",
+    "metadata": {
+      "name": name,
+      "annotations": {
+        "tilt.dev/resource": resource,
+      }
+    },
+    "spec": {
+      "text": text,
+      "location": {
+        "componentType": "Resource",
+        "componentID": resource,
+      }
+    }
+  }
+  cmd = {
+    "apiVersion": "tilt.dev/v1alpha1",
+    "kind": "Cmd",
+    "metadata": {
+      "name": "btn-" + name,
+      "annotations": {
+        "tilt.dev/resource": resource,
+        "tilt.dev/log-span-id": 'cmd:' + name,
+      }
+    },
+    "spec": {
+      "args": argv,
+      "dir": config.main_dir,
+      "startOn": {
+        "startAfter": _now(),
+        "uiButtons": [name],
+      },
+    }
+  }
+  local("echo %s | %s apply -f -" % (shlex.quote(str(encode_yaml_stream([button, cmd]))), sys.executable))
```

Comment:
we're going to need an STDIN stream eventually...

in the meantime, it might be better to do this as an local(env=...) so that the input doesn't show up in the logs...?

Reply:
Good call - I did `echo_off=True` so the command doesn't show up since it's not particularly useful (_especially_ with the YAML invisible) as well as using `env` to eliminate the need for `shlex.quote` which should reduce the likelihood of a weird edge case escaping YAML 😬 

(Also did `echo_off=True` + `quiet=True` for the `_now()` helper)